crates/iota-core/src/congestion_tracker.rs

```diff
@@ -222,72 +309,143 @@ impl CongestionTracker {
         clearing_gas_price
     }
 
-    /// Compute a congestion info map from checkpoint's congestion and
-    /// clearing transactions data.
-    fn compute_congestion_info_map(
+    fn get_total_hotness_for_objects(
+        &self,
+        objects: impl Iterator<Item = ObjectID>,
+    ) -> Option<f64> {
+        let mut total_hotness = 0.0;
+
+        for object_id in objects {
+            if let Some(info) = self.get_congestion_info(object_id) {
+                total_hotness += info.hotness;
+            }
+        }
+        Some(total_hotness)
+    }
```

Comment:
The function does not need to return an `Option`, and can be rewritten as follows:

```suggestion
    fn get_total_hotness_for_objects(&self, objects: impl Iterator<Item = ObjectID>) -> f64 {
        objects
            .filter_map(|object_id| self.get_congestion_info(object_id))
            .map(|info| info.hotness)
            .sum()
    }
```